### PR TITLE
Do not fail pulumi policy new on invalid template(s)

### PR DESCRIPTION
**Pull request by aq17**
Extension of https://github.com/pulumi/pulumi/pull/11435, but for `pulumi policy new`. 
Part of #11434

### CODE REVIEW

File: pkg/cmd/pulumi/policy_new.go [hunk 1]
```diff
@@ -337,18 +340,28 @@ func choosePolicyPackTemplate(templates []workspace.PolicyPackTemplate,
 	surveycore.DisableColor = true
 	message := "\rPlease choose a template:"
 	message = opts.Color.Colorize(colors.SpecPrompt + message + colors.Reset)
+	var selectedOption workspace.PolicyPackTemplate
+
+	for {
```
Comment (iwahbe) [inline pkg/cmd/pulumi/policy_new.go:345]: What are we introducing this `for` loop for? I know it mirrors the last PR, but I think that `has` will always `break` out.

Comment (aq17) [reply pkg/cmd/pulumi/policy_new.go:345]: Yeah I'm not actually sure why we need it in the other file as well. I can remove both.

**Review by iwahbe — APPROVED** [review]
Looks good to me.

**Comment by aq17** [comment]
bors merge

**Comment by aq17** [comment]
bors retry

**Pull request merged**